### PR TITLE
[Merged by Bors] - feat(NumberTheory/MulChar): extend API for multiplicative characters

### PR DESCRIPTION
**Pull request by MichaelStollBayreuth**
This adds some more API lemmas for multiplicative characters.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/NumberTheory/MulChar/Lemmas.lean [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+/-- Define the conjugation (`star`) of a multiplicative character by conjugating pointwise. -/
+@[simps!]
+def starComp [StarRing R'] (χ : MulChar R R') : MulChar R R' :=
+  χ.ringHomComp (starRingEnd R')
+
+
+variable [Fintype R] [DecidableEq R]
+
+/-- The values of a multiplicative character on `R` are `n`th roots of unity, where `n = #Rˣ`. -/
+lemma val_mem_rootsOfUnity (a : Rˣ) {χ : MulChar R R'} :
```
Comment (jcommelin) [inline Mathlib/NumberTheory/MulChar/Lemmas.lean:38]: `val` is usually the name component that refers to `Subtype.val`.
I think in this case `apply` is what should be used.

Idem for the `val`s in lemmas below.

Comment (MichaelStollBayreuth) [reply Mathlib/NumberTheory/MulChar/Lemmas.lean:38]: OK, thanks for pointing this out!

File: Mathlib/NumberTheory/MulChar/Lemmas.lean [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    ← coe_equivToUnitHom, ← coe_equivToUnitHom, Units.ext_iff]
+
+
+section Ring
+
+variable {R R' : Type*} [CommRing R] [CommRing R']
+
+/-- Define the conjugation (`star`) of a multiplicative character by conjugating pointwise. -/
+@[simps!]
+def starComp [StarRing R'] (χ : MulChar R R') : MulChar R R' :=
```
[the remark this answers is not in the document]
Comment (MichaelStollBayreuth) [reply Mathlib/NumberTheory/MulChar/Lemmas.lean:31]: That would be `StarMul`, I think. I'm adding this (and use it to simplify some of the following statements).

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**